### PR TITLE
Update Node, TypeScript, Express and other dependencies

### PR DESCRIPTION
**Pull request by lucacavallaro**
Hello 👋

In this branch I made a massive update (with a small refactor) of some of the libraries we use in the project.

Specifically my focus was on:

- `node` and `@types/node` (now the latest LTS, 16.3.1)
- `typescript` (latest version, 4.5)
- `express` and its types (latest version)

plus other minor libraries and tools like `lodash`, `nodemon` and `figlet` :) 

Also I made two small changes in our codebase:
- `interfaces` exported from `src/utils/server.ts` has now a different type (`Array<{ name: string, address: string }>` instead of `{ loopback: "127.0.0.1", name: string}`). This will enable us to collect all IPv4 addresses associated with network interfaces and not only the first one. The first IPv4 address  collected is exported as `serverIpv4Address`.
- I removed the `postinstall` npm script that triggers the `generate:all` script on every `yarn add`. 

I await your feedback on my changes 😄 

(PS. These updates are preparatory for the next SI team' PR, which will update fp-ts and make the mock server easy to reuse by other projects related to IO)

### CODE REVIEW

**Comment by Undermaken** [comment]
Thank you @lucacavallaro 
Before review could you solve the CI errors?
https://app.circleci.com/pipelines/github/pagopa/io-dev-api-server/1139/workflows/648301e8-f4c1-4296-9c84-7dbd64eb62eb/jobs/1147
I guess they are about lint errors.
You can check it out by running this command in your code
`yarn lint`

File: src/utils/server.ts [hunk 1]
```diff
@@ -1,22 +1,36 @@
...
+  allNets.foldL(
+    () => [],
+    nets => nets.filter(net => net.family === "IPv4" && !net.internal)
+  );
+
+export const interfaces = Object.entries(os.networkInterfaces())
+  .filter(([name]) => interestingNetworkInterfaces.has(name))
+  .reduce(
+    (netInteraces, [name, nets]) => {
+      const addresses = getIpv4Addresses(
```
Comment (gunzip) [inline src/utils/server.ts:21]: nit: `O.fromNullable(nets).map(getIpv4Addresses)` would have been better (avoid writing methods that take monadic structures as input)

**Review by gunzip — APPROVED** [review]

**Review by gunzip — APPROVED** [review]

**Comment by gunzip** [comment]
@Undermaken can we merge this one?

**Comment by Undermaken** [comment]
> @Undermaken can we merge this one?

As soon as possible we'll (@pagopa/io-app) do a complete review of this PR

**Review by Undermaken — APPROVED** [review]

**Pull request merged**